src/main/scala/cromwell/binding/WdlExpression.scala

```diff
@@ -82,11 +91,53 @@ object WdlExpression {
     val parseTree = parser.parse_e(tokens, new WdlSyntaxErrorFormatter(terminalMap))
     new WdlExpression(parseTree.toAst)
   }
+
+  def toString(ast: AstNode, highlighter: SyntaxHighlighter = NullSyntaxHighlighter): String = {
```

Comment:
I watched the video and read up on more related stuff... keep in mind that my opinion is still inchoate.

I'm not sure how I really feel about the Reader monad in this context.  I took a good amount of time watching the video and finding other sites to read about it.  In this particular case, the Reader monad would be used by having `toString` return a `Reader[SyntaxFormatter, String]`, essentially delaying execution until that `Reader` is provided with a `SyntaxFormatter`.  To me that seems like it just reshapes the solution a little bit but doesn't appear to me to provide any benefit over passing in the formatter into a function.  Now instead of:

```scala
expr.format(HtmlSyntaxFormatter)
```
It'd just be called like this

```scala
expr.format()(HtmlSyntaxFormatter)
```

The Reader monad seems to also encourage the use of God objects.  In his case he got around it because all he was sharing was a single Connection object.  But what if your logic requires a lot of things?  I suppose you'd bundle those up in a big object that every method has access to.  In our case, there might be a SyntaxFormatter appropriate for the context, a WdlBinding perhaps, a Backend...

I don't suppose it's possible to compose these readers... like `Reader[WdlBinding, Reader[SyntaxFormatter [String]]`.

Also one comment about the video... my eyebrow raised when he said something to the effect of, "... Inversion of control, which is just a pretentious way of saying 'taking an argument'"  (2:33) And the whole crowd laughed.  I was like... whoa whoa, doesn't inversion of control also usually imply that you are not in control, and instead the thing that IS in control is just calling your code?